mmv1/third_party/terraform/services/sql/resource_sql_database_instance_test.go.tmpl

```diff
@@ -3881,6 +3883,318 @@ func TestAccSqlDatabaseInstance_DiskSizeAutoResizeWithDiskSize(t *testing.T) {
 	})
 }
 
+// CustomDiffMock embeds ResourceDiffMock to override HasChange
```

Comment:
ResourceDiffMock closely replicates the behavior of a real Terraform resource diff. Overriding it to modify behavior may result in tests that don't actually correspond to real-world behavior.

If you're doing this because you ran into a bug where the behavior of ResourceDiffMock doesn't match Terraform resource diff behavior, then we should fix ResourceDiff Mock rather than overriding. (I'd recommend not doing this unless you're extremely sure it's the problem.)